addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -1868,14 +1868,15 @@ export function getFontSizeDisplayValue(sel, getCSSVariableValue, convertNumeric
     let remValue;
     if (closestFontSizedEl) {
         const useFontSizeInput = closestFontSizedEl.style.fontSize;
+        const match = useFontSizeInput.match(/^([\d.]+)([a-zA-Z]+)/);
         if (useFontSizeInput) {
             // Use the computed value to always convert to px. However, this
             // currently does not check that the inline font-size is the one
             // actually having an effect (there could be an !important CSS rule
             // forcing something else).
```

Comment:
This comment addressed the fact of using `getComputedStyle`, which you removed, so the comment should go with it. But I'm not sure whether the "however" part of it is still relevant or not.

@Guillaume-gdi Since you're the one who introduced this, does it seem OK to you?

Reply:
Hello :wave: Thanks to ask!
I'm not sure the comment is still relevant.. :thinking: 
But the diff proposal may bring the following issue
If you have the following html:
```html
<style>
    .odoo {
        font-size: 20px !important;
    }
</style>
<p class="odoo" style="font-size: 100px;">Some random text</p>
```
The editor was correctly displaying **20** as value and now with the changes of this PR it shows **100** which is wrong.

To ensure we are doing the good fix, I think we first need to understand why:
- with the same HTML `<p style="font-size: 2.5rem;"> Some text </p>`
- select the element in dev tools
- type `parseFloat(getComputedStyle($0).fontSize)` in the console
=> In website, project, ... all odoo apps (I guess) it show "40"
=> In email marketing it shows "35"

If you are stuck with it, tell me and I will get a deeper look next week with pleasure :slightly_smiling_face:

Reply:
Right, correct, thank you! @visp-odoo I believe you can keep your general approach but apply it to the return of `getComputedStyle` rather than to `useFontSizeInput`. Now what confuses me though @Guillaume-gdi is why we first test that there's an inline font-size defined before testing this. With `getComputedStyle`, we'll always get a value, regardless of whether that `if` is truthy or not. Couldn't we simply remove that `if`?

Reply:
The thing is that in all apps except mass_mailing, the font size selector is applying responsive font sizes. So depending on the size of the viewport, the effective size may be different. For example in the Website app:
- Drop a text on a page
- Select the text and set 48px as font size (applying `h1-fs` class)
- Toggle de mobile view

=> The editor have to show 48px even if for this screen size, the computed font size is lower than that (because it's responsive)

Reply:
I'm getting more confused ^^' I thought you said above that if there was css that changed the font size, that real font size should be displayed. That's what `getComputedStyle` would give you. But now you're saying if the css changes the font size (because of responsivity), we have to ignore it and show the font size that was applied to the element manually. What am I misunderstanding?

A.
```html
<style>
    div {
        font-size: 24px !important;
    }
</style>
<div style="font-size: 48px;">[abc]</div>
```
Is it correct it should show 24? If so, `getComputedStyle` gives that. If not, we have to use `div.style.fontSize`.

B.
```html
<style>
    div {
        font-size: 24px;
    }
</style>
<div style="font-size: 48px;">[abc]</div>
```
Is it correct it should show 48? If so, both `getComputedStyle` and `div.style.fontSize` give that. If not, we have to actually parse stylesheets.

Thanks for your clarifications :-)

Reply:
> I thought you said above that if there was css that changed the font size, that real font size should be displayed.

It depends.. if the css that is changing the font size is the one of our responsive font size class or not, we don't want to show the computed font size for the reason I explained before:
>in the Website app:
>
>- Drop a text on a page
>- Select the text and set 48px as font size (applying h1-fs class)
>- Toggle de mobile view
>=> The editor have to show 48px even if for this screen size, the computed font size is lower than that (because it's >responsive)

So with this
```html
<style>
.h1-fs {
    font-size: calc(1.425rem + 2.1vw);
}
@media (min-width: 1200px) {
    .h1-fs {
        font-size: 3rem;
    }
}
</style>
<div class="h1-fs">ABC</div>
```
48px must always be displayed, whatever the screen size (even if the computed style is different from 48)

> A. .... Is it correct it should show 24? 

Yes

>B ... Is it correct it should show 48 ?

Yes

Reply:
Ok so if I'm understanding correctly, to my cases A and B, we should add a case C:
```html
<style>
    div {
        font-size: 24px;
    }
    @media (min-width: 1200px) {
        div {
            font-size: 12px;
        }
    }
</style>
<div style="font-size: 48px;">[abc]</div>
```
where it should show 48 no matter the screen size, and a case D:
```html
<style>
    div {
        font-size: 24px !important;
    }
    @media (min-width: 1200px) {
        div {
            font-size: 12px !important;
        }
    }
</style>
<div style="font-size: 48px;">[abc]</div>
```
where it should... what?... always show 24 no matter the screen size? Or always 48? But that would be in conflict with spec A.

It seems a bit crazy if what is expected is to always show the computed style _unless it comes from a media query_, but that's what I'm getting from this. The only way I see to achieve that is to parse all the stylesheets but that's massive.

Reply:
For your case D, I suppose it's ok to show the computed font size as there is no particular font size class.
> It seems a bit crazy if what is expected is to always show the computed style unless it comes from a media query

It's not that, it's more "what is expected is to always show the computed style unless it comes from a responsive font size class or a `styleClassesRelatedToFontSize` or a `tagNameRelatedToFontSize`".

If you want we can do a call about that or have a meeting this week IRL to discuss it... It will be easier than here :)

Reply:
Ah so we should be able to temporarily remove any of this kind of class before looking at the computed style?

Yes I guess best we look at this together irl :-D Thursday GR2? Thanks!